CAIPs/caip-25.md

```diff
@@ -25,11 +25,30 @@ The motivation comes from the lack of standardization across blockchains to expo
 
 ## Specification
 
-The session is proposed by a caller and the response by the respondent is used as the baseline for an ongoing session that both parties will persist.
-The properties and authorization scopes that make up the session are expected to be persisted and tracked over time by both parties in a discrete data store, identified by an entropic [identifier][CAIP-171] assigned in the initial response.
-This object gets updated, extended, closed, etc. by successive calls and notifications, each tagged by this identifier.
+### Language
 
-If a respondent (e.g. a wallet) needs to initiate a new session, whether due to user input, security policy, or session expiry reasons, it can simply generate a new session identifier to signal this notification to the calling provider; if a caller needs to initiate a new session, it can do so by sending a new request without a `sessionIdentifier`. In such cases, a respondent (e.g. wallet) may choose to explicitly close all sessions upon generation of a new one from the same origin or identity, or leave it to time-out; maintaining concurrent sessions is discouraged (see Security Considerations).
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written in
+uppercase in this document are to be interpreted as described in [RFC
+2119][]
+
+### Definition
+
+The session is proposed by a caller and the response by the respondent is used as the baseline for an ongoing session between the two parties.
+- When a wallet responds with a success response containing a `sessionId` (an entropic [identifier][CAIP-171]), the properties and authorization scopes that make up the session should be persisted and tracked over the life of the session by both parties in a discrete data store.
+- When the wallet does not provide a `sessionId` in its initial response, the wallet MUST persist and track the properties and authorization scopes that make up the session. The caller is not expected to persist session data. In this case, the wallet MUST implement a method [`wallet_getSession`][CAIP-307] to enable the caller to query for the current status of the session at any time.
+
+After a session is established between wallet and caller, subsequent `wallet_createSession` calls are used to update the properties and authorization scopes of the session.
+- When a `sessionId` is returned in the initial `wallet_createSession` response, subsequent `wallet_createSession` calls MUST either contain the previously used `sessionId` - on the root of the request - in which case that session will be modified, or the caller must generate a new `sessionId` and pass it with the request, in which case a new session is created and the previous session dangles in parallel (until its expiration, if applicable), though maintaining concurrent sessions is discouraged (see Security Considerations).
```

Comment:
Note that in status quo, caller sending a `wallet_createSession` with NO sessionId re-initiates; a WALLET responding to request with a new sessionId (which a caller can only tell is new if they were managing the old one) is how WALLET reinitiates session.

Reply:
I believe we hashed this out in the meeting and I've included the `lifecycle chart` from our meeting notes: https://github.com/ChainAgnostic/CAIPs/blob/b9db980c335bab7c78322f25d8e39a72ba8874f4/BCP/Session%20Lifecycle%20Management.md#lifecycle-equivalence-chart